test/unit/controllers/ChoiceControllerSpec.scala

```diff
@@ -87,23 +87,21 @@ class ChoiceControllerSpec extends ControllerSpec with OptionValues with BeforeA
       }
     }
 
-    "return 400 (BAD_REQUEST)" when {
+    "throw an IllegalArgumentException" when {
 
       "form is incorrect" in {
 
-        val incorrectForm = Json.toJson(Choice("incorrect"))
-
-        val result = controller.submitChoice()(postRequest(incorrectForm))
-
-        status(result) mustBe BAD_REQUEST
+        intercept[IllegalArgumentException] {
+          await(controller.submitChoice()(postRequest(Json.toJson(Choice("incorrect")))))
+        }
       }
     }
 
     "return 303 (SEE_OTHER)" when {
 
       "choice is Arrival" in {
 
-        val arrivalForm = JsObject(Map("choice" -> JsString(Arrival)))
+        val arrivalForm = JsObject(Map("choice" -> JsString(Arrival.value)))
```

Comment:
I am not 100% sure but if you had provided `Format` for `Choice` syntax like
```
Json.obj("choice" -> Arrival)
```
should works

Reply:
It's not working, you need to pass String